### ISSUE TITLE
[icu] Build error on windows with ASAN

### ISSUE THREAD

**Issue by krawq** [microsoft/vcpkg#29132]
**Host Environment**

- OS: Windows 11
- Compiler: msvc 19.34.31937 for x64

**To Reproduce**

Custom triplet ( = x64-windows with just the last two lines to activate asan for DEBUG build)
```
set(VCPKG_TARGET_ARCHITECTURE x64)
set(VCPKG_CRT_LINKAGE dynamic)
set(VCPKG_LIBRARY_LINKAGE dynamic)
set(VCPKG_C_FLAGS_DEBUG "/fsanitize=address")
set(VCPKG_CXX_FLAGS_DEBUG "/fsanitize=address")
```
`./vcpkg install icu:x64-windows-asan`

**Failure logs**

End of build-x64-windows-asan-dbg-out.log:

_LINK : warning LNK4044: unrecognized option '/luser32'; ignored
LINK : warning LNK4044: unrecognized option '/lgdi32'; ignored
LINK : warning LNK4044: unrecognized option '/lwinspool'; ignored
LINK : warning LNK4044: unrecognized option '/lshell32'; ignored
LINK : warning LNK4044: unrecognized option '/lole32'; ignored
LINK : warning LNK4044: unrecognized option '/loleaut32'; ignored
LINK : warning LNK4044: unrecognized option '/lcomdlg32'; ignored
LINK : warning LNK4044: unrecognized option '/ladvapi32'; ignored
LINK : fatal error LNK1171: unable to load mspdbcore.dll (error code: 1455)_

Command line was : 
Makefile:142: update target 'icudtd72.dll' due to: stubdata.o
`LINK.EXE -DLL -Xlinker -Xlinker -Xlinker -LC:/Users/blabla/source/vcpkg/installed/x64-windows-asan/debug/lib -Xlinker -Xlinker -Xlinker -machine:x64 -Xlinker -Xlinker -Xlinker -nologo -Xlinker -Xlinker -Xlinker -debug -Xlinker -Xlinker -Xlinker -INCREMENTAL  -nologo -NOENTRY -IMPLIB:icudtd.lib -out:icudtd72.dll stubdata.o -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -lcomdlg32 -ladvapi32`

Note: No issue with x64-windows or x64-windows-static triplet - Note icu is part of Qt port (qtbase)

**Comment by Cheney-W** [comment]
Missing `ASAN_SYMBOLIZER_PATH=$(VC_ExecutablePath_x64)`

**Comment by Neumann-A** [comment]
One of the reason this happens because ICU builds an executable which will call the linker explicitly with hardcoded flags and those are not coming from vcpkg. I have the same problem with clang-cl and sanitizer settings. However I don't know exactly why crt runtime symbols are missing.

**Comment by bansan85** [comment]
Same problem with openssl.

Building with static linkage some the problem.

The [readme](https://github.com/openssl/openssl/blob/master/INSTALL.md#enable-asan) says: `It will only work when used with gcc or clang and should be used in conjunction with the no-shared option.`

Maybe the issue should be renamed "Build error on Windows with ASAN and dynamic linkage" without package name in the title.

My triplet:

```
set(VCPKG_TARGET_ARCHITECTURE x64)
set(VCPKG_CRT_LINKAGE dynamic)
set(VCPKG_LIBRARY_LINKAGE static)
set(VCPKG_CXX_FLAGS "-fsanitize=address")
set(VCPKG_C_FLAGS "-fsanitize=address")
```

**Comment by AnyOldName3** [comment]
A new comment

**Comment by bansan85** [comment]
I don't think this ticket should be left opened. Sanitizers are known to work only with static linkage.

**Comment by AnyOldName3** [comment]
That's not a documented limitation for MSVC, the compiler discussed here. After a quick look, I don't think that's true of GCC or Clang, either, as GCC doesn't mention anything like that in its Instrumentation Options page, and all Clang says is that static linking of executables is not supported, which is the opposite limitation to only static linkage working.

**Comment by AnyOldName3** [comment]
A new comment
